modules/perception/onboard/component/fusion_camera_detection_component.cc

```diff
@@ -798,6 +804,56 @@ int FusionCameraDetectionComponent::MakeProtobufMsg(
       return cyber::FAIL;
     }
   }
+
+  // write out lanes in ego coordinates
+  apollo::perception::LaneMarkers *lane_markers =
+      obstacles->mutable_lane_marker();
+  for (const auto &lane : lane_objects) {
+    base::LaneLineCubicCurve curve_coord = lane.curve_image_coord;
+    if (lane.curve_car_coord.a == NULL)
+      curve_coord = lane.curve_car_coord;
+
+    if (lane.pos_type == base::LaneLinePositionType::EGO_LEFT) {
```

Comment:
I actually tried switch cases first. But because the pose_type includes 12 classes, and code will not compile without listing all the cases. So I think it might be more concise to use if/else here.

Reply:
You may need to add `default:` to cover types not listed.